site/app/models/GradingSection.php

```diff
@@ -0,0 +1,82 @@
+<?php
+
+namespace app\models;
+
+use app\libraries\Core;
+
+/**
+ * Represents a single section of students or teams, and their graders
+ * Can be either registration or rotating
+ *
+ * @package app\models
+ * @method Gradeable getGradeable()
+ * @method string getName()
+ * @method User[] getGraders()
+ * @method User[] getUsers()
+ * @method Team[] getTeams()
+ */
+class GradingSection extends AbstractModel {
+    /**
+     * @var Gradeable
+     */
+    protected $gradeable;
+    /**
+     * If this is a registration section (false for rotating)
+     * @var bool
```

Comment:
Mark all of the fields accessible by the magic methods with `@property @var`

Reply:
Somehow that makes PhpStorm not understand types... but will do.

Reply:
Maybe try putting the description after the type name?

Reply:
Correction: it makes PhpStorm not understand array types. `@property @var string` it can get but `@property @var User[]` does not work.